### PR TITLE
Nexus moar interceptors

### PR DESCRIPTION
**Pull request by bergundy**
This PR builds on #5194, review only from f3b03315a1840682f3f04c055fcc67c1346a2eaa.

**Decided to scrap the previous PR and merge this one instead, both have been approved.**

The base branch for this work is `nexus`.

## What changed?

Added `NamespaceRateLimitInterceptor`, `ConcurrentRequestLimitInterceptor`, and `RateLimitInterceptor` for Nexus requests.

Use full method names in the interceptors and quotas instead of just the base name.

## Why?

Need to ensure that we enforce the proper limits for these APIs.

## How did you test it?

Unit tests, I needed to mock the limiters and couldn't leverage functional tests.

## Potential risks

There may be some implications unknown to me for changing the quotas to use full method names.

## Is hotfix candidate?

No.

### CODE REVIEW

File: service/frontend/nexus_handler_test.go [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+type mockAuthorizer struct{}
+
+// Authorize implements authorization.Authorizer.
+func (mockAuthorizer) Authorize(ctx context.Context, caller *authorization.Claims, target *authorization.CallTarget) (authorization.Result, error) {
+	return authorization.Result{Decision: authorization.DecisionAllow}, nil
+}
+
+var _ authorization.Authorizer = mockAuthorizer{}
+
+type mockRateLimiter struct {
```
Comment (pdoerner) [inline service/frontend/nexus_handler_test.go:54]: FWIW we do have a generated mock rate limiter: https://github.com/temporalio/temporal/blob/main/common/quotas/rate_limiter_mock.go

You could also probably achieve the same functionality by using a real rate limiter and adjusting the RPS between 0 and 1 depending on whether the test case should rate limit.

File: service/history/configs/quotas.go [hunk 2]
```diff
@@ -37,65 +37,65 @@ const (
...
-		"GetWorkflowExecutionHistory":            1,
-		"GetWorkflowExecutionHistoryReverse":     1,
-		"GetWorkflowExecutionRawHistoryV2":       1,
-		"ForceDeleteWorkflowExecution":           1,
-		"GetDLQTasks":                            1,
-		"DeleteDLQTasks":                         1,
-		"AddTasks":                               1,
-		"ListQueues":                             1,
+		"/temporal.server.api.historyservice.v1.HistoryService/CloseShard":                             1,
+		"/temporal.server.api.historyservice.v1.HistoryService/GetShard":                               1,
```
Comment (pdoerner) [inline service/history/configs/quotas.go:41]: Did you check for other places we are using method names? I saw some quotas tests that use method name that we might want to update just for consistency.

**Review by pdoerner — APPROVED** [review]

File: service/frontend/nexus_handler_test.go (hunk 1 above)
Comment (bergundy) [reply service/frontend/nexus_handler_test.go:54]: Yeah.. I could use that, but it's so simple that I don't feel like I need the autogenerated mock.

File: service/history/configs/quotas.go (hunk 2 above)
Comment (bergundy) [reply service/history/configs/quotas.go:41]: I updated the quotas tests.

**Pull request merged**